### PR TITLE
Adding support for electrolytes with FPhx states

### PR DESCRIPTION
**Pull request by andrewlee94**
## Summary/Motivation:
This PR builds of some previous work to extend support for FPhx and FcTP states in electrolytes systems. Also fixed a bug in some flow expressions and added some additional features for switching mass/mol basis.

## Changes proposed in this PR:
-
-

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: idaes/generic_models/properties/core/generic/generic_property.py [hunk 1]
```diff
@@ -832,14 +836,16 @@ def define_metadata(cls, obj):
         # TODO : Need to fix to have methods for things that may or may not be
         # created by state var methods
         obj.add_properties(
-            {'flow_mol': {'method': None},
+            {'flow_mol': {'method': 'flow_mol'},
```
Comment (eslickj) [inline idaes/generic_models/properties/core/generic/generic_property.py:839]: Not totally sure, but doesn't the flow\_mol method here create a component with the same name, so would replace itself?

**Review by eslickj — CHANGES_REQUESTED** [review]
Noted one little thing, otherwise looks good as far as I can tell.

**Review by TimBartholomew — APPROVED** [review]
LGTM. My only comment is since the tests for all the bases are similar, it may make sense and would be easier to maintain to have a test harness rather than duplicating the tests in each test file.

**Comment by andrewlee94** [comment]
@eslickj Sorry, missed the underscore. I just pushed the fix. @TimBartholomew could you re-review?

**Review by eslickj — APPROVED** [review]

**Review by TimBartholomew — APPROVED** [review]

**Pull request merged**